docs/operations/clean-metadata-store.md

```diff
@@ -0,0 +1,104 @@
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+When you delete a datasource from Apache Druid, some records related to the datasource may remain in the metadata store including:
+
+- audit records
+- supervisor records
+- rule records
+- compaction configuration records
+- datasource records created by supervisors
+
+If you have a high datasource churn rate, meaning you frequently create and delete many short-lived datasources, the leftover records can start to fill your metadata store and cause performance issues. To maintain metadata store performance in this case, you can configure Apache Druid to automatically remove records associated with deleted datasources from the metadata store.
+
+## Automated cleanup strategies
+There are several cases when you should consider automated cleanup of the metadata related to deleted datasources:
+- Proactively, if you know you have many high-churn datasources. For example you have scripts that create and delete supervisors regularly.
+- If you have issues with the hard disk for your metadata database filling up.
+- If you run into performance issues with the metadata database. For example API calls are very slow or fail to execute.
+
+Do not use the metadata store automated cleanup features if you have requirements to retain metadata records. For example, you have compliance requirements to keep audit records. In these cases, you should come up with an alternate method to preserve the audit metadata while freeing up your active metadata store.
+
+## Configure automated metadata cleanup
+Automated cleanup only removes records for deleted datasources. You can configure cleanup on a per-table basis as follows:
+ - `druid.coordinator.kill.*.on` enables cleanup for a particular metadata table.
+ - `druid.coordinator.kill.*.period` defines the frequency in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) for the cleanup job to check for and delete eligible records.
+ - `druid.coordinator.kill.*.durationToRetain` defines the retention period in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) after creation that the entity becomes eligible for deletion. For example to retain records for 30 days from their creation date: `P30D`.
+
+ For full configuration details, see [Metadata management](../configuration/index.md#metadata-management).
+
+The following are eligible for automatic cleanup:
+- `druid_audit`: Audit records for deleted datasources become eligible for deletion when the `durationToRetain` time has passed since their creation.
+- `druid_supervisors`: Supervisor records become eligible for deletion when:
+
+    - the supervisor is terminated.
+    - the `durationToRetain` time has passed since their creation.
+- `druid_rules`: Rule records become eligible for deletion when:
+
+    - no segments exist for the datasource.
```

Comment:
no segments exist might be too vague. It is when segments are `killed` by the kill task. Meaning that no segments (including both used and unused) exist

Reply:
Might be worth mentioning about kill task as that is required (to remove segments) for rules to become eligible for cleanup